Code/Framework/AzToolsFramework/AzToolsFramework/Manipulators/BoxManipulatorRequestBus.h

```diff
@@ -26,11 +26,19 @@ namespace AzToolsFramework
         virtual AZ::Vector3 GetDimensions() = 0;
         //! Set the X/Y/Z dimensions of the box shape/collider.
         virtual void SetDimensions(const AZ::Vector3& dimensions) = 0;
+        //! @deprecated Because non-uniform scale effects can be complex, it is recommended to separately use
```

Comment:
So I think this comment can be tweaked a little bit as I found it a bit confusing. It seems like the reason this function is deprecated (over using GetWorldTM, GetScale and GetCurrentLocalTransform) is because the implementation of the bus on EditorBoxShapeComponent and EditorColliderComponent is calling TransformNormalizedScale after calculating the world transform of the actual box.
Is this intended? I would say either 
1. rename this function to GetCurrentTransformNoScale.
2. leave the name unchanged but also not extract the scale of the transform, so the non-uniformed scale is already part of the current transform.

Reply:
Since we moved to the quaternion representation for Transform, it cannot handle non-uniform scale. The way NUS works with collider components, you have to do the local transform of the collider, then NUS, then the world transform, which is impossible to combine into a single transform.